pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -412,6 +412,17 @@ class Timestamp(_Timestamp):
                 )
 
         elif ts_input is _no_input:
+            # GH 31200
+            # When year, month or day is not given, we call the datetime
+            # constructor to make sure we get the same error message
+            # since Timestamp inherits datetime
+            if year is None:
```

Comment:
Let's just leverage the `datetime` call below when it's assigned to `ts_input`.

Maybe something like:

```
datetime_kwargs = {'hour': hours or 0, 'minute': minute or 0, ...}
if year is not None:
    datetime_kwargs['year'] = year
if month is not None:
    ...
ts_input = datetime(**datetime_kwargs)
```

Reply:
this is a great idea, I have adapted the code. Thanks